---
title: "TIMs Evolution"
author: "Ahmed Aquib"
date: "2025-08-14"
output:
  powerpoint_presentation: default
  ioslides_presentation: default
editor_options: 
  markdown: 
    wrap: 64
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
BiocManager::install(c("msa", "Biostrings", "seqinr", "ape"))
devtools::install_github("YuLab-SMU/ggmsa")
BiocManager::install("DECIPHER")
library(ggplot2)
library(dplyr)
library(Biostrings)
library("msa")
library("DECIPHER")
library(seqinr)
library(ape)
```

## TIM4

-   TIMD4_HUMAN
-   T-cell immunoglobulin and mucin domain-containing protein 4
-   378 AA in length
-   PS binding receptor expressed on APCs

## TIM4 Domain and Evolution

-   Igv Domain containing protein
-   Orthologs with significant homology present in vertebrates (42%
    Homology with Zebra Fish timd4; Alliance of Genome)
-   No significant homology with invertebrates (C. elegans, and
    Drosophila; Blast)
-   Igv Domain containing proteins are present in invertebrates

## Multiple Sequence Analysis – Species List

**Great Apes:** *Homo sapiens* – Human; *Pan troglodytes* – Common
chimpanzee; *Pan paniscus* – Bonobo; *Gorilla gorilla* – Western
gorilla; *Pongo abelii* – Sumatran orangutan; *Nomascus leucogenys* –
Northern white-cheeked gibbon\
**Old World Monkeys:** *Macaca mulatta* – Rhesus macaque; *Papio anubis*
– Olive baboon\
**New World Monkeys:** *Cebus capucinus* – White-faced capuchin;
*Saimiri boliviensis* – Bolivian squirrel monkey; *Callithrix jacchus* –
Common marmoset\
**Strepsirrhines:** *Lemur catta* – Ring-tailed lemur

## A quick look at the length of CDNA

```{r input data, echo = TRUE}
#Input fasta file save as dna_seqs
dna_seqs <- readDNAStringSet("Sequences/Timd4_Raw_Primate.fasta", format = "fasta")

df <- data.frame(
    name = sub(" .*", "", names(dna_seqs)),
    length = width(dna_seqs) - 1000
)

ggplot(df, aes(x = length, y = reorder(name, length))) +
    geom_col(fill = "indianred1") +
    geom_text(aes(label = length), size = 3.5) + 
    labs(x = "Length above 1000 bp", y = "Species") +
    theme_minimal(base_size = 12)

```

## Multiple Seqence Alignment

```{r codon_aware_alignment, include=FALSE}
aln_codon <- AlignTranslation(dna_seqs, processors = 4)
writeXStringSet(aln_codon, "primates_codon_aln.fasta")
```

```{r alignment, echo=TRUE}
print(aln_codon)
```
## Phylogenetic Tree

```{r phylogenetic_tree}
aln_seqinr <- read.alignment("primates_codon_aln.fasta", format = "fasta")
d<-dist.alignment(aln_seqinr, "identity")
Tree<-nj(d)
Tree_rooted <- root(Tree, outgroup = "Lemur_catta", resolve.root = TRUE)
plot(Tree_rooted, no.margin = TRUE, cex = 0.9)

```

